pkg/rollup/rollup.go

```diff
@@ -0,0 +1,133 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package rollup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+type DownsampleResolution struct {
+	label      string
+	resolution util.DayDuration
+	retention  util.DayDuration
+}
+
+// Parse parses the rollup resolution string and returns true if it respects the format of downsampling resolution,
+// which is a string of comma separated label:resolution:retention.
+// Example: short:5m:90d,long:1h:395d
+func Parse(resolutionStr string) ([]DownsampleResolution, error) {
```

Comment:
This code could go away if we use defined structs in the Downsampling configuration.